### PR TITLE
fix(web): Debug appendTrailingSlash; simplify React route

### PR DESCRIPTION
**Pull request by RyanSkonnord**
~~Apologies for the elaborate kludge.~~ Background:
* https://github.com/getsentry/sentry/pull/25470
* https://getsentry.atlassian.net/browse/ER-587
* https://getsentry.atlassian.net/browse/ER-663

**Update:** No longer a kludge! 🎉

### CODE REVIEW

**Review by markstory — APPROVED** [review]
Seems reasonable to me.

**Review by JoshFerge — COMMENTED** [review]
does this run into the same issue as before with the failed acceptance test b.c. of our react router removing the query string?

**Comment by RyanSkonnord** [comment]
> does this run into the same issue as before with the failed acceptance test b.c. of our react router removing the query string?

That is correct. The integration test fails whenever it catches a `*/releases/*` path -- that is, the same problem with the `[\w-]+` solution, which I inadvertently reproduced by dropping "releases" into the list.

In order for this "bandaid" solution to proceed, I'd want to do some combination of:
1. forego covering "releases" in the fix;
2. do a quick pass to ensure that it won't break other addresses with query strings;
3. fix the deferred bug in the React routing.

I figure "(1 AND 2) OR 3" is necessary.

**Comment by RyanSkonnord** [comment]
@JoshFerge All right, the underlying problem is now fixed for realsies. Care to re-approve?

**Review by JoshFerge — APPROVED** [review]
🔥

File: static/app/routes.tsx [hunk 1]
```diff
@@ -34,7 +34,8 @@ import SettingsWrapper from 'app/views/settings/components/settingsWrapper';
 const appendTrailingSlash: EnterHook = (nextState, replace) => {
   const lastChar = nextState.location.pathname.slice(-1);
   if (lastChar !== '/') {
-    replace(nextState.location.pathname + '/');
+    const pathname = nextState.location.pathname + '/';
+    replace(pathname + nextState.location.search + nextState.location.hash);
```
Comment (markstory) [inline static/app/routes.tsx:38]: :clap:

**Pull request merged**